README.md

```diff
@@ -1,102 +1,262 @@
 
-```coffee
-  ... TODO ...
-  parse_xml
-  parse_assert
-  createAuthRequest
-```
+Takes the following arguments:
+- `IdP` - [IdP](#IdentityProvider)
+- `options`
+  - `relay_state` - SAML relay state.
+  - `auth_context` - Specifies `AuthnContextClassRef`.  This can also be configured on the [SP](#ServiceProvider).
+  - `nameid_format` - Format for Name ID.  This can also be configured on a per-method basis.
+  - `force_authn` - Boolean that allows forcing re-authentication of users even if the user has a SSO session with the IdP.  This can also be configured on the [IdP](#IdentityProvider) or [SP](#ServiceProvider).
+  - `sign_get_request` - Boolean, if true signs the request.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+- `cb(error, url, request_id)` - Callback called with the login URL and ID of the request.
+
+
+<a name="redirect_assert" />
+##### redirect_assert(IdP, options, cb)
+Gets a user object if the login attempt is valid, inflates the response before parsing it.
+
+Takes the following arguments:
+- `IdP` - [IdP](#IdentityProvider)
+- `options`
+  - `request_body` - Object with a `SAMLResponse` or `SAMLRequest`.
+  - `allow_unencrypted_assertion` - Boolean, when set to true allows unencrypted assertions.  This can also be configured on the [IdP](#IdentityProvider) or [SP](#ServiceProvider).
+- `cb(error, response)` - Callback called with the request response
+
+
+<a name="post_assert" />
+##### post_assert(IdP, options, cb)
+Gets a user object if the login attempt is valid.
+
+Takes the following arguments:
+- `IdP` - [IdP](#IdentityProvider)
+- `options`
+  - `request_body` - Object with a `SAMLResponse` or `SAMLRequest`.
+  - `allow_unencrypted_assertion` - Boolean, when set to true allows unencrypted assertions.  This can also be configured on the [IdP](#IdentityProvider) or [SP](#ServiceProvider).
+- `cb(error, response)` - Callback called with the request response.
+
+
+<a name="create_logout_request_url" />
+##### create_logout_request_url(IdP, options, cb)
+Creates a request URL to log out the user.
+
+Takes the following arguments:
+- `IdP` - [IdP](#IdentityProvider).  Note: Can pass `sso_logout_url` instead of IdP.
+- `options`
+  + `name_id` - Format for Name ID.  This can also be configured on a per-method basis.
+  + `session_index` - Session index to use for creating logout request.
+  + `allow_unencrypted_assertion` - Boolean, when set to true allows unencrypted assertions.  This can also be configured on the [IdP](#IdentityProvider) or [SP](#ServiceProvider).
+  + `sign_get_request` - Boolean, if true signs the request.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+  + `relay_state` - SAML relay state.
+- `cb(error, url, request_url)` - Callback called with the logout request url.
+
+
+<a name="create_logout_response_url" />
+##### create_logout_response_url(IdP, options, cb)
+Creates a URL to confirm a successful logout.
```

Comment:
I don't understand this. My impression is this creates a url to send the user to after a logout is done, but where/how would I use this?